language_formatters_pre_commit_hooks/pretty_format_yaml.py

```diff
@@ -14,6 +15,28 @@
 from six import text_type
 
 
+def process_single_document(document, yaml):
+    """Pretty format one document containing YAML or primitive (non-YAML) text.
+    Call this function for each doc within a multi-document .yaml file.
+
+    Args:
+        document (str): Original document content.
+        yaml: YAML library instance.
+
+    Returns:
+        Pretty-formatted content (str).
+    """
+    content = yaml.load(document)
+    if isinstance(content, (list, dict)):
+        pretty_output = StringIO()
+        yaml.dump(content, pretty_output)
+        return pretty_output.getvalue()
+    elif document:
+        # do not disturb primitive content (unstructured text)
+        return str(document)
+    return None
```

Comment:
I would rather return document directly as it is already of type `str` (docstring).
Additionally returning the document if evaluate to `True` means that an empty document (empty string) would be removed from the output documents (according to the around line 80)

```suggestion
    # do not disturb primitive content (ie. unstructured text)
    return document
```

Reply:
I did this deliberately because otherwise it inserts blank documents, resulting in things like this:
```yaml
---
---
```
At this point, I would rather merge what I have and let you make any functional changes you like. I don't have time to test all the variations.

Reply:
++ Created #5 to track this.